src/Service/BookManager.php

```diff
@@ -83,38 +84,36 @@ public function updateBookLocation(Book $book, \SplFileInfo $file): Book
         return $book;
     }
 
+    public function createBookWithoutMetadata(\SplFileInfo $file): Book
+    {
+        $book = new Book();
+
+        $book->setTitle($file->getBasename());
+        $book->setChecksum($this->fileSystemManager->getFileChecksum($file));
+        $book->addAuthor('unknown');
+
+        $book->setExtension($file->getExtension());
+
+        $book->setBookPath('');
+        $book->setBookFilename('');
+
+        return $this->updateBookLocation($book, $file);
+    }
+
     /**
      * @return MetadataType
      *
      * @throws \Exception
      */
     public function extractEbookMetadata(\SplFileInfo $file): array
     {
-        try {
-            if (!Ebook::isValid($file->getRealPath())) {
-                throw new \RuntimeException('Could not read ebook'.$file->getRealPath());
-            }
+        if (!Ebook::isValid($file->getRealPath())) {
+            throw new BookExtractionException('Invalid eBook', $file->getRealPath());
+        }
 
-            $ebook = Ebook::read($file->getRealPath());
-            if (!$ebook instanceof Ebook) {
-                throw new \RuntimeException('Could not read ebook');
-            }
-        } catch (\Throwable) {
-            $ebook = null;
-
-            return [
-                'title' => $file->getFilename(),
-                'authors' => [new BookAuthor('unknown')], // BookAuthor[] (`name`: string, `role`: string)
-                'main_author' => new BookAuthor('unknown'), // ?BookAuthor => First BookAuthor (`name`: string, `role`: string)
-                'description' => null, // ?string
-                'publisher' => null, // ?string
-                'publish_date' => null, // ?DateTime
-                'language' => null, // ?string
-                'tags' => [], // string[] => `subject` in EPUB, `keywords` in PDF, `genres` in CBA
-                'serie' => null, // ?string => `calibre:series` in EPUB, `series` in CBA
-                'serie_index' => null, // ?int => `calibre:series_index` in EPUB, `number` in CBA
-                'cover' => null, //  ?EbookCover => cover of book
-            ];
+        $ebook = Ebook::read($file->getRealPath());
```

Comment:
same here, even if isValid, it happened to throw a Throwable

Reply:
indeed, done